File: 2/solutions/code.go
package main

import "fmt"

func main() {
    limit := 4000000
    prevTerm := 1
    currentTerm := 2
    sumEvenTerms := 0

    for currentTerm <= limit {
        if currentTerm%2 == 0 {
            sumEvenTerms += currentTerm
        }

        nextTerm := prevTerm + currentTerm
        prevTerm = currentTerm
        currentTerm = nextTerm
    }

    fmt.Println(sumEvenTerms)
}

File: 2/solutions/code.py
limit = 4000000
prev_term = 1
current_term = 2
sum_even_terms = 0

while current_term <= limit:
    if current_term % 2 == 0:
        sum_even_terms += current_term

    next_term = prev_term + current_term
    prev_term = current_term
    current_term = next_term

print(sum_even_terms)
